# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/python-maturin/hotfix-ia64.patch.ia64
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- maturin-1.14.1/src/target/mod.rs.vanilla	2026-06-19 07:03:23.000000000 +0200
+++ maturin-1.14.1/src/target/mod.rs	2026-08-01 17:09:11.218527019 +0200
@@ -103,6 +103,7 @@
     Sparc64,
     Sparcv9,
     LoongArch64,
+    Ia64,
 }
 
 impl Arch {
@@ -128,6 +129,7 @@
             Arch::Sparc64 => "sparc64",
             Arch::Sparcv9 => "sparcv9",
             Arch::LoongArch64 => "loongarch64",
+            Arch::Ia64 => "ia64",
         }
     }
 }
@@ -159,6 +161,7 @@
             Arch::Wasm32 => "wasm32",
             Arch::S390X => "s390x",
             Arch::LoongArch64 => "loongarch64",
+            Arch::Ia64 => "ia64",
         }
     }
 }
@@ -193,6 +196,7 @@
             Arch::Mips,
             Arch::Sparc64,
             Arch::LoongArch64,
+            Arch::Ia64,
         ],
         Os::Windows => &[Arch::X86, Arch::X86_64, Arch::Aarch64],
         Os::Macos => &[Arch::Aarch64, Arch::X86_64],
@@ -343,6 +347,7 @@
             Architecture::Sparc64 => Arch::Sparc64,
             Architecture::Sparcv9 => Arch::Sparcv9,
             Architecture::LoongArch64 => Arch::LoongArch64,
+            Architecture::Ia64 => Arch::Ia64,
             unsupported => bail!("The architecture {} is not supported", unsupported),
         };
 
@@ -427,6 +432,7 @@
             Arch::Sparc64 => "sparc64",
             Arch::Sparcv9 => "sparcv9",
             Arch::LoongArch64 => "loongarch64",
+            Arch::Ia64 => "ia64",
         }
     }
 
@@ -528,7 +534,8 @@
             | Arch::Mips
             | Arch::Powerpc
             | Arch::Sparc64
-            | Arch::Sparcv9 => PlatformTag::Linux,
+            | Arch::Sparcv9
+            | Arch::Ia64 => PlatformTag::Linux,
         }
     }
 
@@ -545,7 +552,8 @@
             | Arch::Mips64
             | Arch::Sparc64
             | Arch::Sparcv9
-            | Arch::LoongArch64 => 64,
+            | Arch::LoongArch64
+            | Arch::Ia64 => 64,
             Arch::Armv5teL
             | Arch::Armv6L
             | Arch::Armv7L
